hudi-common/src/main/java/org/apache/hudi/common/util/SerializationUtils.java

```diff
@@ -121,50 +116,16 @@ Object deserialize(byte[] objectData) {
 
     public Kryo newKryo() {
 
-      Kryo kryo = new KryoBase();
+      Kryo kryo = new Kryo();
       // ensure that kryo doesn't fail if classes are not registered with kryo.
       kryo.setRegistrationRequired(false);
       // This would be used for object initialization if nothing else works out.
-      kryo.setInstantiatorStrategy(new org.objenesis.strategy.StdInstantiatorStrategy());
+      kryo.setInstantiatorStrategy(new Kryo.DefaultInstantiatorStrategy(new StdInstantiatorStrategy()));
       // Handle cases where we may have an odd classloader setup like with libjars
       // for hadoop
       kryo.setClassLoader(Thread.currentThread().getContextClassLoader());
       return kryo;
     }
 
-    private static class KryoBase extends Kryo {
-      @Override
-      protected Serializer newDefaultSerializer(Class type) {
-        final Serializer serializer = super.newDefaultSerializer(type);
-        if (serializer instanceof FieldSerializer) {
-          final FieldSerializer fieldSerializer = (FieldSerializer) serializer;
-          fieldSerializer.setIgnoreSyntheticFields(true);
-        }
-        return serializer;
-      }
-
-      @Override
-      protected ObjectInstantiator newInstantiator(Class type) {
-        return () -> {
-          // First try reflectasm - it is fastest way to instantiate an object.
-          try {
-            final ConstructorAccess access = ConstructorAccess.get(type);
-            return access.newInstance();
-          } catch (Throwable t) {
-            // ignore this exception. We may want to try other way.
-          }
-          // fall back to java based instantiation.
-          try {
-            final Constructor constructor = type.getConstructor();
-            constructor.setAccessible(true);
-            return constructor.newInstance();
-          } catch (NoSuchMethodException | IllegalAccessException | InstantiationException
-              | InvocationTargetException e) {
-            // ignore this exception. we will fall back to default instantiation strategy.
-          }
-          return super.getInstantiatorStrategy().newInstantiatorOf(type).newInstance();
```

Comment:
I agree that the `newInstance()` seems like creating a new serializer instance every time if the above conditions are not triggered. And also I think Kyro should be faster than the java default.
What about the user-defined custom payload? Do we need to register it somewhere?

Reply:
Hi @garyli1019, hudi put `HoodieRecord` to kryo, the payload data inside has been coverted to byte[] data, see `BaseAvroPayload#BaseAvroPayload`. So It doesn't affect anything if we register payload class.

Reply:
What I curious about was what you mentioned in the ticket:
```
kryo.register(HoodieKey.class, new HoodieKeySerializer());
kryo.register(GenericData.Record.class, new GenericDataRecordSerializer());
kryo.register(HoodieRecord.class, new HoodieRecordSerializer());
kryo.register(HoodieRecordLocationSerializer.class, new HoodieRecordLocationSerializer());
kryo.register(OverwriteWithLatestAvroPayload.class, new OverwriteWithLatestPayloadSerializer());
```
Where this is done?

Reply:
Get your point, these serializers are not necessary now. 
1. Custom serializers are need when there are special requirements for serialization and deserialization. 
2. We use `InstantiatorStrategy`, which is also efficient. kryo-creation [1] 
    ```
    kryo.setInstantiatorStrategy(new DefaultInstantiatorStrategy(new StdInstantiatorStrategy()));
    ```
3. If we do that currently, it will affect many places.

[1] https://github.com/EsotericSoftware/kryo#object-creation

Reply:
@lamber-ken to gary's point , we also need to consider this case

>What about the user-defined custom payload? Do we need to register it somewhere?

classes extending BaseAvroPayload may be okay. but we allow users to implement their own payload classes.. if they don't have data converted as `byte[]`, do we need to register them etc?